src/editors/typescript/thingworxTypescriptGenerator.ts

```diff
@@ -165,7 +165,14 @@ export class ThingworxToTypescriptGenerator {
             }
             // now generate the service definition, as well as jsdocs
             classDefinition += this.generateJDocWithContent(serviceJsDocElements.join("\n"))
-            classDefinition += `${service.name} (params: ${entityName}.${service.name}Params): ${this.getTypescriptBaseType(outputMetadata)};\n`;
+            if(serviceParamList.length > 0) {
+                // Service has params, reference the interface generated above
+                classDefinition += `${service.name}(params: ${entityName}.${service.name}Params): ${this.getTypescriptBaseType(outputMetadata)};\n`;
+            } else {
+                // If the service has no parameters, declare to overloads for the method, one with no params, and one with an empty object
```

Comment:
Another option here is a single method with an optional parameter e.g. `${service.name}(params?: Record<any, never>)`